### PR TITLE
Now core_pages as controller helper method

### PR DESCRIPTION
**Pull request by araslanov-e**
## What type of PR is this? (check all applicable)
- [x] Refactor
- [ ] Feature
- [ ] Bug Fix

## Description
core_pages as controller helper method
## Added to documentation?
  - [ ] docs.dev.to
  - [ ] readme
  - [X] no documentation needed

### CODE REVIEW

**Comment by araslanov-e** [comment]
How do you like this?

**Comment by rhymes** [comment]
Hi @araslanov-e  I'm personally on the fence on this, I understand why you did it but although application_helper might not be the best place to have a central registry of which controllers are "core" I don't think the solution is to put this information in a method inside the controller itself.

Also, if you go this route you need to have a method `def core_page?` (which should be singular being inside a single class that represents a page) inside EVERY controller you have in the app. You can't have some controllers that return `true` and other controllers that have no method at all, you will get a NoMethodError if by any chance that template is processed, which being the base template... it will definitely happen.

I think this change needs a little more thought, what do you think?

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -70,4 +72,10 @@ def set_no_cache_header
   def touch_current_user
     current_user.touch
   end
+
+  private
+
+  def core_pages?
```
Comment (araslanov-e) [inline app/controllers/application_controller.rb]: @rhymes this method common for alll controllers

Comment (rhymes) [reply app/controllers/application_controller.rb]: Sorry, don't know how I missed it

**Comment by benhalpern** [comment]
I'm a bit on the fence but I think this should work.

Basically the method is designed to tell us which parts of the app get high traffic and should therefore have tighter constraints in terms of performance budget etc.

But it's kind of gotten a bit unwieldy, and I'm wondering what the future should be. But in the meantime this change is _probably_ a good thing.

We'll decide on this soon.

**Pull request closed without merging**